lib/grid.rb

```diff
@@ -1,28 +1,51 @@
 require_relative 'orientation'
+require_relative 'ship'
 
 class Grid
 
-  attr_reader :ships_placed
+  attr_reader :ships_placed, :ships_sunk
 
   def initialize
     @ships_placed = {}
+    @ships_sunk = []
   end
 
   def mark_ship_positions(x, y, ship, orientation)
     @ships_placed[ship] = occupied_cells(x, y, ship.length, orientation)
   end
 
+  def ship?(position)
+     all_occupied_cells = @ships_placed.values.inject {|sum,array| sum + array}
+     all_occupied_cells.include?(position)
+  end
+
+  def ship_on(position)
+    @ships_placed.each do |key, value|
+      if value.include?(position)
+        return key
+      end
+    end
+    nil
+  end
+
+  def add_sunk_ship(ship)
+    if ship.sunk?
+      @ships_sunk.push(ship)
+    end
+  end
+
+  def end_game?
+    @ships_placed.length == @ships_sunk.length
+  end
+
+  private
+
   def occupied_cells(x, y, ship_length, orientation)
-    if orientation == "horizontal"
+    if orientation == "h"
```

Comment:
It might be better to use a symbol instead of a string here. And the previous naming was better! One letter variables or strings are usually not a good idea.